yt/geometry/_selection_routines/selector_object.pxi

```diff
@@ -404,6 +405,100 @@ cdef class SelectorObject:
                             total += mask[i, j, k]
         return total
 
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.cdivision(True)
+    def fill_mask(self, gobj):
+        # This is for an irregular grid.  We make no assumptions about
+        # the shape of the dds values, except that we will regularize to the
+        # max dimension.
+        cdef np.ndarray[np.uint8_t, ndim=3, cast=True] child_mask
+        child_mask = gobj.child_mask
+        cdef np.ndarray[np.uint8_t, ndim=3] mask
+        cdef int dim[3]
+        _ensure_code(gobj.cell_widths[0])
+        _ensure_code(gobj.cell_widths[1])
+        _ensure_code(gobj.cell_widths[2])
+        _ensure_code(gobj.LeftEdge)
+        _ensure_code(gobj.RightEdge)
+        cdef np.ndarray[np.float64_t, ndim=1] oleft_edge = gobj.LeftEdge.d
+        cdef np.ndarray[np.float64_t, ndim=1] oright_edge = gobj.RightEdge.d
+        cdef np.ndarray[np.float64_t, ndim=1] ocell_width
+        cdef int i, n = 0
+        cdef np.float64_t left_edge[3]
+        cdef np.float64_t right_edge[3]
+        cdef np.float64_t **dds
+        dds = <np.float64_t**> malloc(sizeof(np.float64_t*)*3)
+        for i in range(3):
+            dim[i] = gobj.ActiveDimensions[i]
+            left_edge[i] = oleft_edge[i]
+            right_edge[i] = oright_edge[i]
+            dds[i] = <np.float64_t *> malloc(sizeof(np.float64_t) * dim[i])
+            ocell_width = gobj.cell_widths[i]
+            for j in range(dim[i]):
+                dds[i][j] = ocell_width[j]
+        mask = np.zeros(gobj.ActiveDimensions, dtype='uint8')
+        # Check for the level bounds
+        cdef np.int32_t level = gobj.Level
+        # We set this to 1 if we ignore child_mask
+        cdef int total
+        total = self.fill_mask_selector(left_edge, right_edge,
+                                        dds, dim, child_mask,
+                                        mask, level)
+        for i in range(3):
+            free(dds[i])
+        free(dds)
+        if total == 0: return None
+        return mask.astype("bool")
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.cdivision(True)
+    cdef int fill_mask_selector(self, np.float64_t left_edge[3],
+                                np.float64_t right_edge[3],
+                                np.float64_t **dds, int dim[3],
+                                np.ndarray[np.uint8_t, ndim=3, cast=True] child_mask,
+                                np.ndarray[np.uint8_t, ndim=3] mask,
+                                int level):
+        cdef int i, j, k
+        cdef int total = 0, this_level = 0
+        cdef np.float64_t pos[3]
+        if level < self.min_level or level > self.max_level:
+            return 0
+        if level == self.max_level:
+            this_level = 1
+        cdef np.float64_t *offsets[3]
+        cdef np.float64_t c = 0.0
+        cdef np.float64_t tdds[3]
```

Comment:
what does `tdds` stands for ? (I'm already not sure I understand what `dds` means, though I think I get what it's used for)

Reply:
I would guess it is `temporary dds`?

Reply:
yup, `temporary dds`!